Std/Lean/CoreM.lean

```diff
@@ -0,0 +1,43 @@
+/-
+Copyright (c) 2023 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Lean.CoreM
+
+/-!
+# Additional functions using `CoreM` state.
+-/
+
+open Lean
+
+/-- Count the number of heartbeats used during a monadic function. -/
+-- See also `Lean.withSeconds`
+def Lean.withHeartbeats [Monad m] [MonadLiftT BaseIO m] (x : m α) : m (α × Nat) := do
+  let start ← IO.getNumHeartbeats
+  let r ← x
+  let finish ← IO.getNumHeartbeats
+  return (r, (finish - start) / 1000)
```

Comment:
The division by 1000 seems inappropriate here

Reply:
Ah, okay. There's this annoying question of at what point to divide by 1000. Since even after dividing by 1000 its reasonably fine grained (i.e. any tactic invocation however small will take at least 10s?), and there is potential confusion between the user facing "kiloheartbeats" (as used in options) and the internal heartbeats, my instinct had been to minimise(?) confusion by dividing as soon as possible.

Reply:
My rationale is that there is nothing about the name of this function that suggests it would be in different units than `getNumHeartbeats`. It's in moon units anyway, I don't see why not to use it as is.